apps/src/studio/studio.js

```diff
@@ -4330,6 +4393,18 @@ var checkFinished = function () {
     return true;
   }
 
+  if (hasRequiredForSuccess) {
+    if (requiredForSuccess.achieved) {
+      Studio.message = null;
+      Studio.result = ResultType.SUCCESS;
+      return true;
+    } else {
+      // Not meeting these is not reason for immediate failure in itself, but they do
+      // establish a custom error message.
+      Studio.message = requiredForSuccess.message;
```

Comment:
Does this mean there are ways we can succeed but still have Studio.message set? Do we care that we won't get any logging for these different failure types (i.e. if we wanted to run a query on the server to see how frequently people fail to hit removedItemCount, we have no way to do so with this design)